File: Week 3/spectrum.py
def getColour(wave):
    if wave >= 750:
        return "unknown"
    elif wave >= 620:
        return "red"
    elif wave >= 590:
        return "orange"
    elif wave >= 570:
        return "yellow"
    elif wave >= 495:
        return "green"
    elif wave >= 450:
        return "blue"
    elif wave >= 380:
        return "violet"
    else:
        return "unknown"

wave = input("enter the wavelength: ")
print("The colour for the light with lambda = " + wave + " nm is " + getColour(float(wave)))

File: Week 3/spectrum.c
#include <stdio.h>
char getColour(float lambda);

int main(){
  printf("Lambda (nm): ");
  float lambda;
  scanf("%f", &lambda);
  char symbol = getColour(lambda);
  printf("The colour for lambda = %.1f nm is \'%c\'\n", lambda, symbol);
}

char getColour (float lambda){
  if (lambda >= 750){
    return 'u';
  } else if (lambda >= 620){
    return 'R';
  } else if (lambda >= 590){
    return 'O';
  } else if (lambda >= 570){
    return 'Y';
  } else if (lambda >= 495){
    return 'G';
  } else if (lambda >= 450){
    return 'B';
  } else if (lambda >= 380){
    return 'V';
  } else {
    return 'u';
  }
}